src/shims/time.rs

```diff
@@ -132,7 +134,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         })?;
         this.write_scalar(
             Scalar::from_i64(qpc),
-            &this.deref_operand(lpPerformanceCount_op)?.into(),
+            &this.deref_operand_as(lpPerformanceCount_op, this.machine.layouts.u64)?.into(),
```

Comment:
I am not entirely sure this is worth it for primitive types... this adds quite a bit of churn, and makes the Miri code a lot harder to write. It can be hard to find out which Rust type some C API type corresponds to. Currently when we get it wrong, we get an error saying the size is wrong, which is pretty helpful. With this change we might not get an error at all, just clobbering bytes beyond what we are supposed to be writing to.

So if we do this I think it is important that `deref_operand_as` also checks that the layout given by the shim has the same size and the layout given by the user.

Reply:
> So if we do this I think it is important that deref_operand_as also checks that the layout given by the shim has the same size and the layout given by the user.

From what I understand this is explicitly something you did not want. So for now it's probably better to just stick to `deref_operand` for primitive types. We will already detect wrong sizes there anyway since we detect when the scalar written to a place does not match the size of the place. (And we deliberately changed that from an ICE to a UB error so that Miri could use this for its shims.)